<?xml version="1.0" encoding="utf-8"?><testsuite errors="0" failures="2" name="pytest" skips="0" tests="52" time="1.009"><testcase classname="valid8.tests.test_core" file="valid8/tests/test_core.py" line="8" name="test_validate_nominal_builtins" time="0.0027000904083251953"><system-out>hello
</system-out></testcase><testcase classname="valid8.tests.test_core" file="valid8/tests/test_core.py" line="32" name="test_validate_nominal_custom_3_styles" time="0.00486302375793457"><system-out>hello
</system-out></testcase><testcase classname="valid8.tests.test_core" file="valid8/tests/test_core.py" line="73" name="test_validate_empty" time="0.001443624496459961"></testcase><testcase classname="valid8.tests.test_core" file="valid8/tests/test_core.py" line="82" name="test_validate_none_wrong_notnone" time="0.0010917186737060547"></testcase><testcase classname="valid8.tests.test_core" file="valid8/tests/test_core.py" line="92" name="test_validate_none_enforce" time="0.0029497146606445312"><system-out>hello
</system-out></testcase><testcase classname="valid8.tests.test_core" file="valid8/tests/test_core.py" line="113" name="test_validate_none_pytypes" time="0.013755321502685547"><system-out>hello
</system-out></testcase><testcase classname="valid8.tests.test_core" file="valid8/tests/test_core.py" line="134" name="test_validate_none_is_allowed" time="0.0019562244415283203"><system-out>hello
hello
</system-out></testcase><testcase classname="valid8.tests.test_core" file="valid8/tests/test_core.py" line="146" name="test_validate_name_error" time="0.001150369644165039"></testcase><testcase classname="valid8.tests.test_core" file="valid8/tests/test_core.py" line="155" name="test_validate_not_not_all" time="0.0031845569610595703"><system-out>hello
</system-out></testcase><testcase classname="valid8.tests.test_core" file="valid8/tests/test_core.py" line="185" name="test_validate_or" time="0.002527952194213867"><system-out>hello
hello
hello
hello
</system-out></testcase><testcase classname="valid8.tests.test_core" file="valid8/tests/test_core.py" line="220" name="test_validate_xor" time="0.002328157424926758"><system-out>hello
hello
hello
</system-out></testcase><testcase classname="valid8.tests.test_core" file="valid8/tests/test_core.py" line="254" name="test_validate_and" time="0.0034933090209960938"><system-out>hello
hello
</system-out></testcase><testcase classname="valid8.tests.test_core" file="valid8/tests/test_core.py" line="290" name="test_decorate_manually" time="0.0015015602111816406"></testcase><testcase classname="valid8.tests.test_mini_lambda" file="valid8/tests/test_mini_lambda.py" line="13" name="test_evaluator_iterable" time="0.0010495185852050781"></testcase><testcase classname="valid8.tests.test_mini_lambda" file="valid8/tests/test_mini_lambda.py" line="28" name="test_evaluator_iterator" time="0.0014040470123291016"></testcase><testcase classname="valid8.tests.test_mini_lambda" file="valid8/tests/test_mini_lambda.py" line="49" name="test_evaluator_iterator_iterable" time="0.0009822845458984375"></testcase><testcase classname="valid8.tests.test_mini_lambda" file="valid8/tests/test_mini_lambda.py" line="60" name="test_evaluator_iterable_iterator_and_comparison" time="0.0013756752014160156"></testcase><testcase classname="valid8.tests.test_mini_lambda" file="valid8/tests/test_mini_lambda.py" line="72" name="test_evaluator_comprehension" time="0.0010890960693359375"></testcase><testcase classname="valid8.tests.test_mini_lambda" file="valid8/tests/test_mini_lambda.py" line="81" name="test_evaluator_iterable_any" time="0.0012848377227783203"></testcase><testcase classname="valid8.tests.test_mini_lambda" file="valid8/tests/test_mini_lambda.py" line="95" name="test_evaluator_iterable_all" time="0.0010344982147216797"></testcase><testcase classname="valid8.tests.test_mini_lambda" file="valid8/tests/test_mini_lambda.py" line="110" name="test_evaluator_repr" time="0.0014231204986572266"></testcase><testcase classname="valid8.tests.test_mini_lambda" file="valid8/tests/test_mini_lambda.py" line="127" name="test_evaluator_complex_1" time="0.0019485950469970703"></testcase><testcase classname="valid8.tests.test_mini_lambda" file="valid8/tests/test_mini_lambda.py" line="138" name="test_evaluator_str" time="0.0012581348419189453"></testcase><testcase classname="valid8.tests.test_mini_lambda" file="valid8/tests/test_mini_lambda.py" line="154" name="test_evaluator_bytes" time="0.0016527175903320312"></testcase><testcase classname="valid8.tests.test_mini_lambda" file="valid8/tests/test_mini_lambda.py" line="171" name="test_evaluator_sizeof" time="0.0009503364562988281"></testcase><testcase classname="valid8.tests.test_mini_lambda" file="valid8/tests/test_mini_lambda.py" line="189" name="test_evaluator_comparable" time="0.0009970664978027344"></testcase><testcase classname="valid8.tests.test_mini_lambda" file="valid8/tests/test_mini_lambda.py" line="209" name="test_evaluator_comparable_normal_function_first" time="0.0010416507720947266"></testcase><testcase classname="valid8.tests.test_mini_lambda" file="valid8/tests/test_mini_lambda.py" line="221" name="test_evaluator_comparable_both_evaluators" time="0.0012977123260498047"></testcase><testcase classname="valid8.tests.test_mini_lambda" file="valid8/tests/test_mini_lambda.py" line="234" name="test_evaluator_hashable" time="0.0014293193817138672"></testcase><testcase classname="valid8.tests.test_mini_lambda" file="valid8/tests/test_mini_lambda.py" line="250" name="test_evaluator_truth_testable" time="0.0013513565063476562"></testcase><testcase classname="valid8.tests.test_mini_lambda" file="valid8/tests/test_mini_lambda.py" line="266" name="test_evaluator_truth_testable_not" time="0.0013573169708251953"></testcase><testcase classname="valid8.tests.test_mini_lambda" file="valid8/tests/test_mini_lambda.py" line="282" name="test_evaluator_attribute" time="0.0018079280853271484"></testcase><testcase classname="valid8.tests.test_mini_lambda" file="valid8/tests/test_mini_lambda.py" line="301" name="test_evaluator_nonexistent_attribute_2" time="0.0015895366668701172"></testcase><testcase classname="valid8.tests.test_mini_lambda" file="valid8/tests/test_mini_lambda.py" line="352" name="test_evaluator_container" time="0.0014657974243164062"></testcase><testcase classname="valid8.tests.test_mini_lambda" file="valid8/tests/test_mini_lambda.py" line="369" name="test_evaluator_sized" time="0.0013277530670166016"></testcase><testcase classname="valid8.tests.test_mini_lambda" file="valid8/tests/test_mini_lambda.py" line="383" name="test_evaluator_sized_compared" time="0.0015349388122558594"></testcase><testcase classname="valid8.tests.test_mini_lambda" file="valid8/tests/test_mini_lambda.py" line="406" name="test_evaluator_reversible" time="0.0012929439544677734"></testcase><testcase classname="valid8.tests.test_mini_lambda" file="valid8/tests/test_mini_lambda.py" line="418" name="test_evaluator_mapping" time="0.001222372055053711"></testcase><testcase classname="valid8.tests.test_mini_lambda" file="valid8/tests/test_mini_lambda.py" line="444" name="test_evaluator_numeric" time="0.003225088119506836"></testcase><testcase classname="valid8.tests.test_mini_lambda" file="valid8/tests/test_mini_lambda.py" line="499" name="test_evaluator_int_convertible" time="0.0011560916900634766"></testcase><testcase classname="valid8.tests.test_mini_lambda" file="valid8/tests/test_mini_lambda.py" line="513" name="test_evaluator_long_convertible" time="0.001107931137084961"><failure message="NameError: name &apos;long&apos; is not defined">def test_evaluator_long_convertible():
        &quot;&quot;&quot; Long convertible Object : tests that `long` raises the appropriate exception and that equivalent Long()
        works &quot;&quot;&quot;
    
        s = InputVar(float)
    
        # with pytest.raises(NotImplementedError):
        #     int(s)
    
&gt;       to_long = long(s)
E       NameError: name &apos;long&apos; is not defined

valid8/tests/test_mini_lambda.py:523: NameError</failure></testcase><testcase classname="valid8.tests.test_mini_lambda" file="valid8/tests/test_mini_lambda.py" line="528" name="test_evaluator_float_convertible" time="0.001272439956665039"></testcase><testcase classname="valid8.tests.test_mini_lambda" file="valid8/tests/test_mini_lambda.py" line="543" name="test_evaluator_complex_convertible" time="0.0013306140899658203"></testcase><testcase classname="valid8.tests.test_mini_lambda" file="valid8/tests/test_mini_lambda.py" line="559" name="test_evaluator_oct_convertible" time="0.0013077259063720703"></testcase><testcase classname="valid8.tests.test_mini_lambda" file="valid8/tests/test_mini_lambda.py" line="574" name="test_evaluator_index_slice" time="0.0010488033294677734"></testcase><testcase classname="valid8.tests.test_readme" file="valid8/tests/test_readme.py" line="7" name="test_first_example_enforce" time="0.006343364715576172"><system-out>you did it !
you did it !
</system-out></testcase><testcase classname="valid8.tests.test_readme" file="valid8/tests/test_readme.py" line="41" name="test_first_example_pytypes" time="0.006016254425048828"><failure message="pytypes.exceptions.InputTypeError: 
  valid8.tests.test_readme.build_house
  called with incompatible types:
Expected: Tuple[str, Real, Union[Integral, NoneType], Boolean]
Received: Tuple[str, int, int, int]">def test_first_example_pytypes():
        # for type checking
        from valid8 import Boolean
        from numbers import Real, Integral
        from typing import Optional
        from pytypes import typechecked
    
        # for value checking
        from valid8 import validate, minlens, gt
    
        @typechecked
        @validate(name=minlens(0),
                  surface=gt(0))
        def build_house(name: str,
                        surface: Real,
                        nb_floors: Optional[Integral] = 1,
                        with_windows: Boolean = False):
            print(&apos;you did it !&apos;)
    
&gt;       build_house(&apos;test&apos;, 12, 2)  # validation OK

valid8/tests/test_readme.py:61: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../../virtualenv/python3.5.3/lib/python3.5/site-packages/pytypes/typechecker.py:825: in checker_tp
    prop_getter or auto_prop_getter, specs)
../../../virtualenv/python3.5.3/lib/python3.5/site-packages/pytypes/typechecker.py:673: in _checkfunctype
    _raise_typecheck_error(msg, False, check_val, tpch, argSig, func)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

msg = &apos;\n  valid8.tests.test_readme.build_house\n  called with incompatible types:\nExpected: Tuple[str, Real, Union[Integral, NoneType], Boolean]\nReceived: Tuple[str, int, int, int]&apos;
is_return = False, value = (&apos;test&apos;, 12, 2, 1)
received_type = typing.Tuple[str, int, int, int]
expected_type = typing.Tuple[str, numbers.Real, typing.Union[numbers.Integral, NoneType], valid8.utils_typing.Boolean]
func = &lt;function test_first_example_pytypes.&lt;locals&gt;.build_house at 0x7f5eee257510&gt;

    def _raise_typecheck_error(msg, is_return=False, value=None, received_type=None,
                expected_type=None, func=None):
        if pytypes.warning_mode:
            import traceback
            tb = traceback.extract_stack()
            off = util._calc_traceback_list_offset(tb)
            cat = pytypes.ReturnTypeWarning if is_return else pytypes.InputTypeWarning
            warn_explicit(msg, cat, tb[off][0], tb[off][1])
    # 		if not func is None:
    # 			warn_explicit(msg, cat, func.__code__.co_filename,
    # 					func.__code__.co_firstlineno, func.__module__)
    # 		else:
    # 			warn(msg, pytypes.ReturnTypeWarning)
        else:
            if is_return:
                raise pytypes.ReturnTypeError(msg)
            else:
&gt;               raise pytypes.InputTypeError(msg)
E               pytypes.exceptions.InputTypeError: 
E                 valid8.tests.test_readme.build_house
E                 called with incompatible types:
E               Expected: Tuple[str, Real, Union[Integral, NoneType], Boolean]
E               Received: Tuple[str, int, int, int]

../../../virtualenv/python3.5.3/lib/python3.5/site-packages/pytypes/type_util.py:1446: InputTypeError</failure></testcase><testcase classname="valid8.tests.test_readme" file="valid8/tests/test_readme.py" line="73" name="test_custom_validators" time="0.0036165714263916016"></testcase><testcase classname="valid8.tests.test_validators_collections" file="valid8/tests/test_validators_collections.py" line="7" name="test_validate_is_in" time="0.003097057342529297"><system-out>hello
</system-out></testcase><testcase classname="valid8.tests.test_validators_collections" file="valid8/tests/test_validators_collections.py" line="20" name="test_validate_subset_superset" time="0.0032880306243896484"><system-out>hello
</system-out></testcase><testcase classname="valid8.tests.test_validators_collections" file="valid8/tests/test_validators_collections.py" line="44" name="test_validate_on_all" time="0.0022423267364501953"><system-out>hello
</system-out></testcase><testcase classname="valid8.tests.test_validators_collections" file="valid8/tests/test_validators_collections.py" line="55" name="test_validate_on_each" time="0.00234222412109375"><system-out>hello
</system-out></testcase></testsuite>